client/src/main/java/io/paradoxical/cassieq/api/client/CassieqCredentials.java

```diff
@@ -67,8 +60,7 @@ static CassieqCredentials signedQueryString(String queryAuth) {
                           .url(newUrlBuilder.build())
                           .build();
         };
-
     }
 
-    Request authorize(Request request);
+    Request authorize(Request request) throws NoSuchAlgorithmException, InvalidKeyException;
```

Comment:
NoSuchAlgorithm should be swallowed and surfaced as Assertion. doesn't make sense to throw that if the user doesn't specify the algorithm.